lib/rspec/support/encoded_string.rb

```diff
@@ -33,17 +49,55 @@ def to_s
 
         private
 
+        # Encoding Exceptions:
+        #
+        # Raised by Encoding and String methods:
+        #   Encoding::UndefinedConversionError:
+        #     when a transcoding operation fails
+        #     if the String contains characters invalid for the target encoding
+        #     e.g. "\x80".encode('UTF-8','ASCII-8BIT')
+        #     and "\x80".encode('UTF-8','ASCII-8BIT', undef: :replace, replace: '<undef>')
+        #     # => '<undef>'
+        #   Encoding::CompatibilityError
+        #    when Enconding.compatbile?(str1, str2) is false
```

Comment:
this is technically incorrect.  as written, it's when `nil`.  I'll revise if everything else is ok.

I was confusing [enc.ascii_compatible?](https://github.com/ruby/ruby/blob/ca24e581ba/encoding.c#L493)

```c
* Returns whether ASCII-compatible or not.
 *
 *   Encoding::UTF_8.ascii_compatible?     #=> true
 *   Encoding::UTF_16BE.ascii_compatible?  #=> false
```

with [rb_enc_check](https://github.com/ruby/ruby/blob/ca24e581ba/encoding.c#L844)

```c
rb_enc_check(VALUE str1, VALUE str2)
{
    rb_encoding *enc = rb_enc_compatible(str1, str2);
    if (!enc)
	rb_raise(rb_eEncCompatError, "incompatible character encodings: %s and %s",
```

where [`enc_compatible` is `Encoding.compatible?(str1,str2)](https://github.com/ruby/ruby/blob/ca24e581ba/encoding.c#L1194)